src/deadline/client/cli/_groups/manifest_group.py

```diff
@@ -228,28 +238,113 @@ def manifest_download(
+        logger=logger,
+    )
+    logger.json(dataclasses.asdict(output))
 
 
 @cli_manifest.command(
     name="upload",
-    help="BETA - Uploads a job attachment manifest file to a Content Addressable Storage's Manifest store. If calling via --s3-cas-path, it is recommended to use with --profile for a specific AWS profile with CAS S3 bucket access.",
+    help="BETA - Uploads a job attachment manifest file to a Content Addressable Storage's Manifest store. If calling via --s3-cas-path, it is recommended to use with --profile for a specific AWS profile with CAS S3 bucket access. Check exit code for success or failure.",
 )
 @click.argument("manifest_file")
 @click.option("--profile", help="The AWS profile to use.")
-@click.option("--s3-cas-path", help="The path to the Content Addressable Storage root.")
+@click.option("--s3-cas-uri", help="The URI to the Content Addressable Storage S3 bucket and root.")
+@click.option(
+    "--s3-manifest-prefix", help="Prefix subpath in the manifest folder to upload the manifest."
+)
 @click.option(
-    "--farm-id", help="The AWS Deadline Cloud Farm to use. Alternative to using --s3-cas-path."
+    "--farm-id", help="The AWS Deadline Cloud Farm to use. Alternative to using --s3-cas-uri."
 )
 @click.option(
-    "--queue-id", help="The AWS Deadline Cloud Queue to use. Alternative to using --s3-cas-path."
+    "--queue-id", help="The AWS Deadline Cloud Queue to use. Alternative to using --s3-cas-uri."
 )
 @click.option("--json", default=None, is_flag=True, help="Output is printed as JSON for scripting.")
 @_handle_error
 def manifest_upload(
     manifest_file: str,
-    s3_cas_path: str,
+    s3_cas_uri: str,
+    s3_manifest_prefix: str,
     json: bool,
     **args,
 ):
-    raise NotImplementedError("This CLI is being implemented.")
+    # Input checking.
+    if not manifest_file or not os.path.isfile(manifest_file):
+        raise NonValidInputError(f"Specified manifest {manifest_file} does not exist. ")
+
+    # Where will we upload the manifest to?
+    required: set[str] = set()
+    if not s3_cas_uri:
+        required = {"farm_id", "queue_id"}
+
+    config: Optional[ConfigParser] = _apply_cli_options_to_config(required_options=required, **args)
+
+    # Logger
+    logger: ClickLogger = ClickLogger(is_json=json)
+
+    bucket_name: str = ""
+    session: boto3.Session = api.get_boto3_session(config=config)
+    if not s3_cas_uri:
+        farm_id = config_file.get_setting("defaults.farm_id", config=config)
+        queue_id = config_file.get_setting("defaults.queue_id", config=config)
+
+        deadline = api.get_boto3_client("deadline", config=config)
+        queue = deadline.get_queue(
+            farmId=farm_id,
+            queueId=queue_id,
+        )
+        queue_ja_settings: JobAttachmentS3Settings = JobAttachmentS3Settings(
+            **queue["jobAttachmentSettings"]
+        )
+        bucket_name = queue_ja_settings.s3BucketName
+        cas_path = queue_ja_settings.rootPrefix
```

Comment:
nit - is there specific reasons to rename `rootPrefix` to cas_pass? Could we directly access them via JobAttachmentS3Settings instance?

Reply:
None, purely a variable naming thing.